Dracwyrm's collection of updates.

dracwyrm opened the pull request:
profiles/arch/powerpc: mask unkeyworded use flags
--Mask the use flags for packages that aren't keyworded yet.

media-libs/ilmbase: Revision bump to 2.2.0-r1 and change to CMake
--The CMake build system is the future of this project so switch to it now.
--This does fix some errors linking to previously installed versions.
--Added a patch to fix some post release bugs.

media-libs/openexr: Revision bump to 2.2.0-r2 and change to CMake
--The CMake build system is the future of this project so switch to it now.
--This does fix some errors linking to previously installed versions.
--Added a patch to fix some post release bugs.
--Also, the CMake build installs more header files, some of which are needed by OpenImageIO.

media-libs/ctl: version bump to 1.52
--Fixed depends as well.

media-libs/Field3D: New package

media-libs/opencollada: version bump to 1.6.58

media-libs/openimageio: version bumps to 1.6.18 and 1.7.16
--1.6.18 is the last release of the 1.6 series, so add it in for compatibility with older software.
--1.7 series has been out for awhile, but not in portage.

media-libs/opencolorio: add latest upstream fixes.
--This is a patch to add all the latest fixes since they seem to have stopped releasing new versions.
--There are new yaml fixes that are not in the patch that's in portage.
--There are numerous other fixes which makes this patch release nice.

media-libs/ptex: revision bump to 2.1.28-r1

media-libs/opensubdiv: Version bump to 3.3 and remove ati-drivers

media-libs/SeExpr: New package

media-libs/partio: New package

media-libs/osl: New package

media-gfx/openexr_viewers: Revision bump to 2.2.0-r1
--The version in portage has outdated depends.
--Aslo, switched to the CMake build system.

media-gfx/openvdb: version bump to 4.0.2

media-gfx/blender: version bump to 2.78c

@gentoo/proxy-maint

dracwyrm commented:
I just pushed an update that adds a legacy symlink so all packages depending on IlmBase and OpenEXR should work with no trouble, given that they are supposed to work with v2.2.0. Though, upstream would like all applications needing IlmBase and OpenEXR to use their new format, so will drop patch in future versions.

SoapGentoo commented on media-gfx/blender/blender-2.72b-r4.ebuild:
```diff
@@ -246,8 +247,7 @@ src_install() {
 		dodoc -r "${CMAKE_USE_DIR}"/doc/doxygen/html/*
```
1. delete the `src_test() { :; }`
2. Add `RESTRICT="test"` after `REQUIRED_USE`
3. Add a comment why

SoapGentoo replied:
`dodoc -r "${CMAKE_USE_DIR}"/doc/doxygen/html/.` (Avoid unnecessary globs)

SoapGentoo commented on media-gfx/blender/blender-2.72b-r4.ebuild:
```diff
@@ -246,8 +247,7 @@ src_install() {
+	cmake-utils_src_install
 
 	# fix doc installdir
 	docinto "html"
```
drop the quotes

SoapGentoo commented on media-gfx/blender/blender-2.78c.ebuild:
```diff
@@ -0,0 +1,298 @@
+
+DESCRIPTION="3D Creation/Animation/Publishing System"
+HOMEPAGE="http://www.blender.org"
+
```
remove this empty line

SoapGentoo commented on media-gfx/blender/blender-2.78c.ebuild:
```diff
@@ -0,0 +1,298 @@
+	)"
+
+PATCHES=( "${FILESDIR}/${PN}-fix-install-rules.patch"
+	  "${FILESDIR}/${PN}-2.78-eigen-3.3.1.patch" )
```
```
PATCHES=(
	"${FILESDIR}/${PN}-fix-install-rules.patch"
	"${FILESDIR}/${PN}-2.78-eigen-3.3.1.patch"
)
```

SoapGentoo commented on media-gfx/blender/blender-2.78c.ebuild:
```diff
@@ -0,0 +1,298 @@
+	[[ ${MERGE_TYPE} != binary ]] && use openmp && tc-check-openmp
+
+	if use doc; then
+		CHECKREQS_DISK_BUILD="4G" check-reqs_pkg_pretend
```
put this in the `[[ ${MERGE_TYPE} != binary ]]` clause

dracwyrm replied:
How do I do that? This should run no matter what build we are doing???

SoapGentoo replied:
```
blender_check_requirements() {
	if [[ ${MERGE_TYPE} != binary ]]; then
		use openmp && tc-check-openmp

		if use doc; then
			CHECKREQS_DISK_BUILD="4G" check-reqs_pkg_pretend
		fi
	fi
}
```

SoapGentoo commented on media-gfx/blender/blender-2.78c.ebuild:
```diff
@@ -0,0 +1,298 @@
+src_configure() {
+	# FIX: forcing '-funsigned-char' fixes an anti-aliasing issue with menu
+	# shadows, see bug #276338 for reference
+	append-flags -funsigned-char
```
this sounds like some undefined behaviour. It's ok for the time being, but please see if you can get it fixed.

dracwyrm replied:
I removed the flag to see what happens. Maybe this was valid for the old version but fixed in the new. Testing.

SoapGentoo commented on media-gfx/blender/blender-2.78c.ebuild:
```diff
@@ -0,0 +1,298 @@
+	cmake-utils_src_install
+
+	# fix doc installdir
+	docinto "html"
```
chuck quotes

SoapGentoo commented on media-gfx/openexr_viewers/openexr_viewers-2.2.0-r1.ebuild:
```diff
@@ -0,0 +1,45 @@
+		$(cmake-utils_use_find_package cg Cg)
+		-DOPENEXR_PACKAGE_PREFIX="${EPREFIX}/usr"
+		-DILMBASE_PACKAGE_PREFIX="${EPREFIX}/usr"
+		-DCMAKE_INSTALL_DOCDIR="share/doc/${PF}"
```
:+1: :+1:

SoapGentoo commented on media-gfx/openexr_viewers/openexr_viewers-2.2.0-r1.ebuild:
```diff
@@ -0,0 +1,45 @@
+		-DCMAKE_INSTALL_DOCDIR="share/doc/${PF}"
+	)
+
+	use cg && append-flags "$(no-as-needed)" # binary-only libCg is not properly linked
```
christ nvidia you suck...

dracwyrm replied:
Nothing I can do about that, either.

SoapGentoo commented on media-gfx/openvdb/files/openvdb-4.0.2-use-pkg-config.patch:
```diff
@@ -0,0 +1,131 @@
+ FIND_PACKAGE ( PackageHandleStandardArgs )
++FIND_PACKAGE ( PkgConfig )
++
++PKG_CHECK_MODULES ( ILMBASE_LIBS REQUIRED IlmBase )
```
finally sanity...

SoapGentoo commented on media-gfx/openvdb/openvdb-4.0.2.ebuild:
```diff
@@ -0,0 +1,68 @@
+
+DESCRIPTION="Libs for the efficient manipulation of volumetric data"
+HOMEPAGE="http://www.openvdb.org"
+
```
remove empty line

SoapGentoo commented on media-gfx/openvdb/openvdb-4.0.2.ebuild:
```diff
@@ -0,0 +1,68 @@
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="+abi3-compat doc python"
+
```
this one too

SoapGentoo commented on media-gfx/openvdb/openvdb-4.0.2.ebuild:
```diff
@@ -0,0 +1,68 @@
+
+PATCHES=( "${FILESDIR}/${P}-use-pkg-config.patch"
+	  "${FILESDIR}/${P}-use-gnuinstalldirs.patch"
+)
```
ditto here too

SoapGentoo commented on media-libs/Field3D/Field3D-1.7.2.ebuild:
```diff
@@ -0,0 +1,27 @@
+
+DESCRIPTION="A library for storing voxel data"
+HOMEPAGE="http://opensource.imageworks.com/?p=field3d"
+
```
ditto

SoapGentoo commented on media-libs/Field3D/files/Field3D-1.7.2-openexr-fixes.patch:
```diff
@@ -0,0 +1,38 @@
++find_package(PkgConfig QUIET)
++if(PKG_CONFIG_FOUND)
++  pkg_check_modules(PC_ILMBASE QUIET IlmBase)
++endif()
```
whats the rationale here? Use pkgconfig as one of a number of ways to discover IlmBase?

dracwyrm replied:
I have no clue. This was an upstream patch. I can change it later and submit something decent upstream.

SoapGentoo commented on media-libs/SeExpr/SeExpr-2.11.ebuild:
```diff
@@ -0,0 +1,36 @@
+
+DESCRIPTION="An embeddable expression evaluation engine"
+HOMEPAGE="http://www.disneyanimation.com/technology/seexpr.html"
+
```
ditto

SoapGentoo commented on media-libs/SeExpr/SeExpr-2.11.ebuild:
```diff
@@ -0,0 +1,36 @@
+IUSE="doc"
+
+RDEPEND="sys-devel/llvm:=
+	media-libs/libpng:=
```
`:0=`

SoapGentoo commented on media-libs/SeExpr/files/SeExpr-2.11-build-fixes.patch:
```diff
@@ -0,0 +1,49 @@
+     set(SIP_QT_TAG -t Qt_4_5_0)  # fix 'sip: QFileIconProvider is undefined'
+ 
+-    set(CMAKE_CXX_FLAGS "-std=c++0x")
++    set(CMAKE_CXX_FLAGS "-std=c++11")
```
this is not how you set C++11 in CMake since 3.1:

```
set(CMAKE_CXX_STANDARD 11)
set(CMAKE_CXX_STANDARD_REQUIRED ON)
set(CMAKE_CXX_EXTENSIONS OFF)
```

SoapGentoo commented on media-libs/ctl/ctl-1.5.2.ebuild:
```diff
@@ -0,0 +1,26 @@
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+inherit cmake-utils
```
free line before inherit

SoapGentoo commented on media-libs/ctl/files/ctl-1.5.2-cmake-fixes.patch:
```diff
@@ -0,0 +1,309 @@
++exec_prefix=@CMAKE_INSTALL_PREFIX@
++libdir=@CMAKE_INSTALL_FULL_LIBDIR@
++includedir=@CMAKE_INSTALL_FULL_INCLUDEDIR@
++CTL_includedir=@CMAKE_INSTALL_FULL_INCLUDEDIR@/CTL
```
nix the `exec_prefix` line - it's an Autoconf implementation detail

SoapGentoo commented on media-libs/ctl/files/ctl-1.5.2-cmake-fixes.patch:
```diff
@@ -0,0 +1,309 @@
++exec_prefix=@CMAKE_INSTALL_PREFIX@
++libdir=@CMAKE_INSTALL_FULL_LIBDIR@
++includedir=@CMAKE_INSTALL_FULL_INCLUDEDIR@
++OpenEXR_includedir=@CMAKE_INSTALL_FULL_INCLUDEDIR@/OpenEXR
```
nix both `prefix` and `exec_prefix`

SoapGentoo commented on media-libs/ilmbase/files/ilmbase-2.2.0-fix-pkgconfig-file.patch:
```diff
@@ -0,0 +1,53 @@
++++ b/IlmBase.pc.cmake	2017-08-17 08:48:29.114556759 +0100
+@@ -0,0 +1,11 @@
++prefix=@CMAKE_INSTALL_PREFIX@
++exec_prefix=@CMAKE_INSTALL_PREFIX@
```
again, nix `prefix` and `exec_prefix`

SoapGentoo commented on media-libs/ilmbase/files/ilmbase-2.2.0-fix-pkgconfig-file.patch:
```diff
@@ -0,0 +1,53 @@
++Description: Base math and exception libraries
++Version: @ILMBASE_VERSION@
++Libs: -L${libdir} -lHalf -lIex@ILMBASE_LIBSUFFIX@ -lIexMath@ILMBASE_LIBSUFFIX@ -lIlmThread@ILMBASE_LIBSUFFIX@ -lImath@ILMBASE_LIBSUFFIX@ -pthread
++Cflags: -pthread -I${OpenEXR_includedir}
```
In general the correct approach for pthreads is:

1. Call
```
set(CMAKE_THREAD_PREFER_PTHREAD ON)
find_package(Threads)
```
2. Then use `@CMAKE_THREAD_LIBS_INIT@` in the pkgconfig template file instead of `-pthread`

but that is only a recommendation, choose whether you want to go this route or not

dracwyrm replied:
I'll change my upstream PR to this and then add it here.

SoapGentoo commented on media-libs/ilmbase/ilmbase-2.2.0-r1.ebuild:
```diff
@@ -0,0 +1,26 @@
+SLOT="0/12" # based on SONAME
+KEYWORDS="~amd64 -arm ~arm64 ~hppa ~ia64 ~ppc ~ppc64 ~sparc ~x86 ~amd64-fbsd ~x86-fbsd ~amd64-linux ~x86-linux ~x64-macos ~x86-solaris"
+
+DOCS=( AUTHORS ChangeLog NEWS README )
```
None of these require defining `DOCS` explicitly

SoapGentoo commented on media-libs/ilmbase/ilmbase-2.2.0-r1.ebuild:
```diff
@@ -0,0 +1,26 @@
+
+PATCHES=( "${FILESDIR}/${P}-post-release-fixes-v20170109.patch"
+	  "${FILESDIR}/${P}-use-gnuinstall-dirs.patch"
+	  "${FILESDIR}/${P}-fix-pkgconfig-file.patch"
```
reofrmat these too (and **never** use spaces for indentation)

SoapGentoo commented on media-libs/opencollada/opencollada-1.6.58.ebuild:
```diff
@@ -0,0 +1,68 @@
+
+KEYWORDS="~amd64 ~ppc64 ~x86"
+
+IUSE="static-libs"
```
`LICENSE`/`SLOT`/`KEYWORDS`/`IUSE` form one block

SoapGentoo commented on media-libs/opencolorio/files/opencolorio-1.0.9-cmake-fixes-v1.patch:
```diff
@@ -0,0 +1,150 @@
+-
+ install(DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/build-html/
+-        DESTINATION ${CMAKE_INSTALL_PREFIX}/share/doc/OpenColorIO/html
++        DESTINATION ${CMAKE_INSTALL_PREFIX}/share/doc/$ENV{P}/html
```
`${CMAKE_INSTALL_DOCDIR}/html` ?

dracwyrm replied:
This was an existing Gentoo patch, but I'll convert it to GNUInstallDirs.

SoapGentoo commented on media-libs/openexr/files/openexr-2.2.0-fix-pkgconfig-file.patch:
```diff
@@ -0,0 +1,39 @@
++++ b/OpenEXR.pc.cmake	2017-08-17 12:15:19.425695228 +0100
+@@ -0,0 +1,14 @@
++prefix=@CMAKE_INSTALL_PREFIX@
++exec_prefix=@CMAKE_INSTALL_PREFIX@
```
same here again

SoapGentoo commented on media-libs/openexr/files/openexr-2.2.0-fix-pkgconfig-file.patch:
```diff
@@ -0,0 +1,39 @@
++Libs: -L${libdir} -lIlmImf@OPENEXR_LIBSUFFIX@ -lIlmImfUtil@OPENEXR_LIBSUFFIX@
++Cflags: -I${OpenEXR_includedir}
++Requires: IlmBase
++Libs.private: -lz
```
`Requires.private: zlib`

SoapGentoo commented on media-libs/openexr/openexr-2.2.0-r2.ebuild:
```diff
@@ -0,0 +1,56 @@
+
+# Only need hard blocker for 2.2.0 series. Can remove on version bump.
+RDEPEND="sys-libs/zlib[${MULTILIB_USEDEP}]
+	 ~media-libs/ilmbase-${PV}:=[${MULTILIB_USEDEP}]
```
are you using spaces to try and indent stuff? dont do that, use only tabs

SoapGentoo commented on media-libs/openimageio/openimageio-1.6.18.ebuild:
```diff
@@ -0,0 +1,99 @@
+	>=media-libs/libwebp-0.2.1:=
+	media-libs/openexr:=
+	media-libs/tiff:0=
+	sys-libs/zlib:0=
```
zlib has no SLOTs, remove this

SoapGentoo commented on media-libs/openimageio/openimageio-1.6.18.ebuild:
```diff
@@ -0,0 +1,99 @@
+	qt4? (
+		dev-qt/qtcore:4=
+		dev-qt/qtgui:4=
+		dev-qt/qtopengl:4=
```
remvoe `=` on all Qt deps

SoapGentoo commented on media-libs/openimageio/openimageio-1.6.18.ebuild:
```diff
@@ -0,0 +1,99 @@
+	[[ -z $mysimd ]] && mysimd="0"
+
+	local mycmakeargs=(
+		-DLIB_INSTALL_DIR="/usr/$(get_libdir)"
```
1. sounds like this also needs `GNUInstallDir`
2. `-DLIB_INSTALL_DIR="${EPREFIX}/usr/$(get_libdir)"`

dracwyrm replied:
I'll make a local patch as upstream won't except 1.6 branch patches anymore.

SoapGentoo commented on media-libs/openimageio/openimageio-1.7.16.ebuild:
```diff
@@ -0,0 +1,122 @@
+	[[ -z $mysimd ]] && mysimd="0"
+
+	local mycmakeargs=(
+		-DLIB_INSTALL_DIR="/usr/$(get_libdir)"
```
ditto

dracwyrm replied:
This I can submit upstream and was going to. :)

SoapGentoo replied:
You dont have to kill yourself - if you get `GNUInstallDirs` into trunk, thats fine, you can keep these vars then

dracwyrm replied:
True. I'll keep them for now. However, 1.6.18 is the last version of 1.6 branch, so I want to make it pretty, because it may be around for awhile as stuff still depends on 1.6.x.

SoapGentoo commented on media-libs/openimageio/openimageio-1.7.16.ebuild:
```diff
@@ -0,0 +1,122 @@
+		-DVERBOSE=ON
+	)
+
+	if [[ ${EPYTHON} == python3* ]]; then
```
`if python_is_python3; then`

SoapGentoo commented on media-libs/opensubdiv/files/opensubdiv-3.3.0-fix-quotes.patch:
```diff
@@ -0,0 +1,218 @@
+ 
+     add_definitions(
+-        -DOPENSUBDIV_VERSION_STRING="${OSD_SONAME}"
++        -DOPENSUBDIV_VERSION_STRING=${OSD_SONAME}
```
have you tried `-DOPENSUBDIV_VERSION_STRING=\\\"${OSD_SONAME}\\\"` without all the following macro stringify hackery

SoapGentoo commented on media-libs/opensubdiv/opensubdiv-3.3.0.ebuild:
```diff
@@ -0,0 +1,62 @@
+DESCRIPTION="An Open-Source subdivision surface library"
+HOMEPAGE="http://graphics.pixar.com/opensubdiv/"
+
+MY_PV="$(replace_all_version_separators '_')"
```
put this before `DESCRIPTION` and then put `SRC_URI` right after `HOMEPAGE`

SoapGentoo commented on media-libs/opensubdiv/opensubdiv-3.3.0.ebuild:
```diff
@@ -0,0 +1,62 @@
+
+LICENSE="ZLIB"
+SLOT="0"
+IUSE="cuda doc examples opencl openmp ptex tbb tutorials"
```
`KEYWORDS` goes after `SLOT`

SoapGentoo commented on media-libs/opensubdiv/opensubdiv-3.3.0.ebuild:
```diff
@@ -0,0 +1,62 @@
+S="${WORKDIR}/OpenSubdiv-${MY_PV}"
+
+PATCHES=( "${FILESDIR}/${P}-fix-quotes.patch"
+	  "${FILESDIR}/${P}-cmake-fixes.patch"
```
weird space indentation

SoapGentoo commented on media-libs/osl/osl-1.8.10.ebuild:
```diff
@@ -0,0 +1,70 @@
+	 >=media-libs/openimageio-1.7.0
+	 dev-libs/pugixml
+	 sys-libs/zlib
+	 partio? ( media-libs/partio )"
```
weird space indents again

SoapGentoo commented on media-libs/partio/partio-1.1.0_p20160426.ebuild:
```diff
@@ -0,0 +1,46 @@
+
+IUSE="doc"
+
+REQUIRED_USE="${PYTHON_REQUIRED_USE}"
```
remvoe the empty line before `IUSE` and `REQUIRED_USE`

SoapGentoo commented on media-libs/ptex/files/ptex-2.1.28-20170213.patch:
```diff
@@ -0,0 +1,173 @@
+ 
+ if (NOT WIN32)
+-    set(CMAKE_CXX_FLAGS "-std=c++98 -Wall -Wextra -pedantic")
++    set(CMAKE_CXX_FLAGS "-std=c++98 -Wall -Wextra -pedantic ${CMAKE_CXX_FLAGS}")
```
wat. Is this package brain-damaged or what? C++98? really?

dracwyrm replied:
I changed it to 11, because it's easy to modify a patch like that, and it seems to work. Will need to make sure it doesn't segfault under testing.

SoapGentoo commented on media-libs/ptex/files/ptex-2.1.28-20170213.patch:
```diff
@@ -0,0 +1,173 @@
+     VERBOSE=1
+diff -purN a/Ptex.spec.in b/Ptex.spec.in
+--- a/Ptex.spec.in	1970-01-01 01:00:00.000000000 +0100
++++ b/Ptex.spec.in	2017-02-13 22:59:10.000000000 +0000
```
do you really want to put this in a patch?

dracwyrm replied:
This is all unmodified upstream. I can remove the spec file section, but then it wouldn't be upstream, but ah well. Apparently it's 1970 at upstream. haha

SoapGentoo commented on media-libs/ptex/ptex-2.1.28-r1.ebuild:
```diff
@@ -0,0 +1,22 @@
+DESCRIPTION="Per-Face Texture Mapping for Production Rendering"
+HOMEPAGE="http://ptex.us/"
+SRC_URI="https://github.com/wdas/ptex/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+LICENSE="BSD"
```
free line before `LICENSE`

SoapGentoo commented on media-libs/ptex/ptex-2.1.28-r1.ebuild:
```diff
@@ -0,0 +1,22 @@
+SRC_URI="https://github.com/wdas/ptex/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+LICENSE="BSD"
+SLOT="0"
+RDEPEND="sys-libs/zlib"
```
free line before `RDEPEND` + `KEYWORDS` goes directly below `SLOT`

SoapGentoo commented on media-libs/ptex/ptex-2.1.28-r1.ebuild:
```diff
@@ -0,0 +1,22 @@
+KEYWORDS="~amd64 ~x86"
+
+PATCHES=( "${FILESDIR}/${P}-20170213.patch"
+	"${FILESDIR}/${P}-allow-custom-build-type.patch" )
```
formatting

SoapGentoo commented on media-libs/ptex/ptex-2.1.28-r1.ebuild:
```diff
@@ -0,0 +1,22 @@
+PATCHES=( "${FILESDIR}/${P}-20170213.patch"
+	"${FILESDIR}/${P}-allow-custom-build-type.patch" )
+
+mycmakeargs=( -DCMAKE_INSTALL_DOCDIR="${EPREFIX}/usr/share/doc/${PF}/html" )
```
`mycmakeargs=( -DCMAKE_INSTALL_DOCDIR=share/doc/${PF}/html )`

dracwyrm commented:
@SoapGentoo I just pushed some fixes to your review comments. I'm working on the remainder now. In case you want to start checking. Also, I fixed some issues with OpenEXR_CTL.pc file. I added in Requires, which cleaned it up a lot! You can see the difference in the diff.

dracwyrm commented:
@SoapGentoo Pushed the latest changes that should take care of all your requests. I guess each patch needs to be clicked on to see if it's correct, as in no files from other packages accidentally included. Also, they should be in order of commit that won't break main repo between commits. I'll double check, but it's good to have an extra set of eyes for that.

dracwyrm commented:
@SoapGentoo Good thing I did some double checking, I forgot a depend in OpenImageIO 1.6.18.

dracwyrm commented:
@SoapGentoo Actually, hold a bit. They may be removing SeExpr support from partio, so then I won't need that package anymore. I need to investigate.

SoapGentoo commented on media-gfx/openvdb/files/openvdb-4.0.2-use-gnuinstalldirs.patch:
```diff
@@ -0,0 +1,83 @@
+-	RUNTIME DESTINATION bin COMPONENT Runtime
+-	ARCHIVE DESTINATION lib COMPONENT Runtime
++	RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR} COMPONENT Runtime
++	ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR} COMPONENT Runtime
```
afaik this is also missing a `LIBRARY` argument (not important, only a Windoze problem)

SoapGentoo commented on media-libs/opencollada/opencollada-1.6.58.ebuild:
```diff
@@ -0,0 +1,69 @@
+		-DUSE_SHARED=ON
+		-DUSE_STATIC=$(usex static-libs)
+		-DUSE_LIBXML=ON
+		-Dsoversion=${MY_SOVERSION}
```
it's fine here, but setting SONAMEs yourself is dodgy, as this should usually be done by upstreams.

SoapGentoo commented on media-libs/opencolorio/opencolorio-1.0.9_p20170719.ebuild:
```diff
@@ -0,0 +1,86 @@
+		)
+	python? ( ${PYTHON_DEPS} )
+	>=dev-cpp/yaml-cpp-0.5
+	dev-libs/tinyxml
```
put unconditional deps before conditional ones (and then only have one tab for the closing parentheses of the `opengl?` clause)

SoapGentoo commented on media-libs/opencolorio/opencolorio-1.0.9_p20170719.ebuild:
```diff
@@ -0,0 +1,86 @@
+	"
+
+# Documentation building requires Python bindings building
+REQUIRED_USE="doc? ( python )"
```
missing `python? ( ${PYTHON_REQUIRED_USE} )`

SoapGentoo commented on media-libs/openimageio/openimageio-1.6.18.ebuild:
```diff
@@ -0,0 +1,101 @@
+	truetype? ( media-libs/freetype:2= )"
+DEPEND="${RDEPEND}"
+
+PATCHES=( ${FILESDIR}/${P}-Use-GNUInstallDirs.patch )
```
`"${FILESDIR}"` (think about spaces in file paths)

SoapGentoo commented on media-libs/partio/partio-1.1.0_p20170829.ebuild:
```diff
@@ -0,0 +1,46 @@
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+
+IUSE="doc"
```
remove free line before

SoapGentoo commented on media-libs/ptex/files/ptex-2.1.28-20170213.patch:
```diff
@@ -0,0 +1,72 @@
+ 
+ if (NOT WIN32)
+-    set(CMAKE_CXX_FLAGS "-std=c++98 -Wall -Wextra -pedantic")
++    set(CMAKE_CXX_FLAGS "-std=c++11 -Wall -Wextra -pedantic ${CMAKE_CXX_FLAGS}")
```
in general, this is the wrong way to set any C++ standard in CMake...

SoapGentoo commented on media-libs/ptex/ptex-2.1.28-r1.ebuild:
```diff
@@ -0,0 +1,25 @@
+	"${FILESDIR}/${P}-allow-custom-build-type.patch"
+)
+
+mycmakeargs=( -DCMAKE_INSTALL_DOCDIR="share/doc/${PF}/html" )
```
this will fail in EAPI 7. You should explicitly install html files to `html/` of `CMAKE_INSTALL_DOCDIR`, not change the canonical docdir such that it works.

dracwyrm commented:
I'm breaking this up into smaller chunks for review. Closing.

devurandom commented:
Could you please link those new PRs back here, so it will be easier for others to find and subscribe to them?

dracwyrm commented:
For IlmBase, OpenEXR, CTL, and OpenEXR_Viewers: https://github.com/gentoo/gentoo/pull/5725
I can only do one PR at a time since there are depends. I figure this is the easiest way to get the whole thing into the Main Repo, because it's less to focus on.

dracwyrm commented:
media-libs/Field3D: PR https://github.com/gentoo/gentoo/pull/5748
media-libs/opencollada: PR https://github.com/gentoo/gentoo/pull/5749
media-libs/opencolorio: PR https://github.com/gentoo/gentoo/pull/5750
media-libs/opensubdiv: PR https://github.com/gentoo/gentoo/pull/5751
media-libs/partio: PR https://github.com/gentoo/gentoo/pull/5752
media-libs/ptex: PR https://github.com/gentoo/gentoo/pull/5753
media-gfx/openvdb: PR https://github.com/gentoo/gentoo/pull/5754

dracwyrm commented:
OpenImageIO, OSL, and Blender are blocked until the above PRs are merged.
